app/models/merchant.rb

```diff
@@ -1,6 +1,16 @@
 class Merchant < ApplicationRecord
   has_many :items
   has_many :invoices
+  has_many :invoice_items, through: :items
   has_many :transactions, through: :invoices
   has_many :customers, through: :invoices
+
+  def self.most_items_sold(quantity)
+    Merchant.
+      select("merchants.*, SUM(invoice_items.quantity) AS most_items").
+      joins(:invoice_items).
+      order("most_items DESC").
+      group("merchants.id").
+      first(quantity)
```

Comment:
I'm not positive, but from what I'm trying it seems that `"?", quantity` type of injection is only available in certain queries like `where` when you're fishing for a particular attribute that the models have. Methods like `first`, `take`, and `limit` only except at most one argument. Let's ask an instructor about whether this is a security flaw or whether we should leave it in when it comes to production.